app/services/mentors/check_training_completion.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module Mentors
+  class CheckTrainingCompletion < BaseService
+    EARLY_ROLL_OUT_COMPLETION_DATE = Date.new(2021, 4, 19)
+
+    def call
+      return unless mentor_profile.mentor?
+
+      set_completion_values!
+    end
+
+  private
+
+    attr_reader :mentor_profile
+
+    def initialize(mentor_profile:)
+      @mentor_profile = mentor_profile
+    end
+
+    def set_completion_values!
+      completion_date, completion_reason = if ero_mentor?
```

Comment:
I think a risk here is we override reasons/dates if something changes (even though we are favouring ERO), maybe we should guard against that if the date/reason already exists on a profile to avoid changes? Also a question elsewhere, do we want to support multiple reasons from the get go?

Reply:
These are always going to work in a given order of precedence so should be fine. Only the ERO one is not triggered by an event and currently you should not be both ERO and declared, and if you were, ERO should trump the declaration.  I don't think this is a problem.  I want to ensure we re-evaluate the completion state when we receive a void/clawback because when the other reasons are introduced we should ensure we're not clearing the completion when some other reason might apply further down the precedence list perhaps.